shelley/chain-and-ledger/executable-spec/src/STS/Ledgers.hs

```diff
@@ -70,7 +71,23 @@ ledgersTransition = do
   let (favs', ready) = Map.partitionWithKey (\s _ -> s > slot) favs
   let avs' = Applications $ apps avs ⨃ (Map.toList . apps $ newAVs avs ready)
   let u''' = UTxOState utxo' dep fee (ppup, aup, favs', avs')
-  pure $ LedgerState u''' dw'' (_txSlotIx ls)
+
+  let ds = _dstate dw''
+      fdms_    = _fdms ds
+      Dms dms_ = _dms ds
+      (fdms', curr) = Map.partitionWithKey (\(s, _) _ -> slot <= s) fdms_
+  let maxSlot = maximum . Set.map fst . Map.keysSet
+  let latestPerGKey gk =
+        ( (maxSlot . Map.filterWithKey (\(_, c) _ -> c == gk)) curr
```

Comment:
can do this with `Relation` 

  maxSlot (curr ▷ Set.singleton gk)

or just 

  Map.filter (== gk) curr

Reply:
much better, thank you!

Reply:
sadly, I don't think this works. we aren't filtering on the range, but on piece of the pair in the domain.